### PR TITLE
additional data for type StopWatch in swagger api

### LINKED ISSUE #12457: Proposal - api - add more details to "type StopWatch" [WIP]
- Gitea version (or commit ref): 96add8c319bd8299bfb8fdd4d98c6f71407e2475
- Git version: 
- Operating system: Linux
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

It is not possible to get more info via swagger API for a "StopWatch" (especially the repository assigned to its issue) after a API GET request `/user/stopwatches`. Actual I am getting this StopWatch type after a request:
```
{
  created        string($date-time)
  issue_index  integer($int64)
}
```
The issue index is assigned to a repo, but repo data is not part of the type/data. I want to change this type to:

```
{
  created | string($date-time)
  issue_index  integer($int64)
  issue_title  string
  repo_name  string
  repo_owner_name  string
}
```

here my fork: https://github.com/no-go/gitea

### Possible problems after that "hack"

- Other Code, that suggests the old/reduced `type StopWatch`
- Security problems: Access Issue data to a StopWatch, but user is not allowed to see the issue/repo anymore

### PR DESCRIPTION
**Pull request by alltagshack**
Fixes #12457

I test it with (Linux, go 1.14.5) `make build` and `make test`:
```
go test  -mod=vendor -tags='sqlite sqlite_unlock_notify' code.gitea.io/gitea code.gitea.io/gitea/cmd code.gitea.io/gitea/contrib/environment-to-ini code.gitea.io/gitea/contrib/fixtures code.gitea.io/gitea/contrib/pr code.gitea.io/gitea/models code.gitea.io/gitea/models/migrations code.gitea.io/gitea/modules/analyze code.gitea.io/gitea/modules/auth code.gitea.io/gitea/modules/auth/ldap code.gitea.io/gitea/modules/auth/oauth2 code.gitea.io/gitea/modules/auth/openid code.gitea.io/gitea/modules/auth/pam code.gitea.io/gitea/modules/auth/sso code.gitea.io/gitea/modules/avatar code.gitea.io/gitea/modules/base code.gitea.io/gitea/modules/cache code.gitea.io/gitea/modules/charset code.gitea.io/gitea/modules/context code.gitea.io/gitea/modules/convert code.gitea.io/gitea/modules/cron code.gitea.io/gitea/modules/emoji code.gitea.io/gitea/modules/eventsource code.gitea.io/gitea/modules/generate code.gitea.io/gitea/modules/git code.gitea.io/gitea/modules/git/pipeline code.gitea.io/gitea/modules/gitgraph code.gitea.io/gitea/modules/graceful code.gitea.io/gitea/modules/highlight code.gitea.io/gitea/modules/httplib code.gitea.io/gitea/modules/indexer/code code.gitea.io/gitea/modules/indexer/issues code.gitea.io/gitea/modules/indexer/stats code.gitea.io/gitea/modules/lfs code.gitea.io/gitea/modules/log code.gitea.io/gitea/modules/markup code.gitea.io/gitea/modules/markup/common code.gitea.io/gitea/modules/markup/csv code.gitea.io/gitea/modules/markup/external code.gitea.io/gitea/modules/markup/markdown code.gitea.io/gitea/modules/markup/mdstripper code.gitea.io/gitea/modules/markup/orgmode code.gitea.io/gitea/modules/metrics code.gitea.io/gitea/modules/migrations code.gitea.io/gitea/modules/migrations/base code.gitea.io/gitea/modules/notification code.gitea.io/gitea/modules/notification/action code.gitea.io/gitea/modules/notification/base code.gitea.io/gitea/modules/notification/indexer code.gitea.io/gitea/modules/notification/mail code.gitea.io/gitea/modules/notification/ui code.gitea.io/gitea/modules/notification/webhook code.gitea.io/gitea/modules/options code.gitea.io/gitea/modules/password code.gitea.io/gitea/modules/pprof code.gitea.io/gitea/modules/private code.gitea.io/gitea/modules/process code.gitea.io/gitea/modules/public code.gitea.io/gitea/modules/queue code.gitea.io/gitea/modules/recaptcha code.gitea.io/gitea/modules/references code.gitea.io/gitea/modules/repofiles code.gitea.io/gitea/modules/repository code.gitea.io/gitea/modules/secret code.gitea.io/gitea/modules/session code.gitea.io/gitea/modules/setting code.gitea.io/gitea/modules/ssh code.gitea.io/gitea/modules/structs code.gitea.io/gitea/modules/svg code.gitea.io/gitea/modules/sync code.gitea.io/gitea/modules/task code.gitea.io/gitea/modules/templates code.gitea.io/gitea/modules/test code.gitea.io/gitea/modules/timeutil code.gitea.io/gitea/modules/upload code.gitea.io/gitea/modules/user code.gitea.io/gitea/modules/util code.gitea.io/gitea/modules/validation code.gitea.io/gitea/modules/webhook code.gitea.io/gitea/routers code.gitea.io/gitea/routers/admin code.gitea.io/gitea/routers/api/v1 code.gitea.io/gitea/routers/api/v1/admin code.gitea.io/gitea/routers/api/v1/misc code.gitea.io/gitea/routers/api/v1/notify code.gitea.io/gitea/routers/api/v1/org code.gitea.io/gitea/routers/api/v1/repo code.gitea.io/gitea/routers/api/v1/settings code.gitea.io/gitea/routers/api/v1/swagger code.gitea.io/gitea/routers/api/v1/user code.gitea.io/gitea/routers/api/v1/utils code.gitea.io/gitea/routers/dev code.gitea.io/gitea/routers/events code.gitea.io/gitea/routers/org code.gitea.io/gitea/routers/private code.gitea.io/gitea/routers/repo code.gitea.io/gitea/routers/routes code.gitea.io/gitea/routers/user code.gitea.io/gitea/routers/user/setting code.gitea.io/gitea/routers/utils code.gitea.io/gitea/services/comments code.gitea.io/gitea/services/externalaccount code.gitea.io/gitea/services/gitdiff code.gitea.io/gitea/services/issue code.gitea.io/gitea/services/mailer code.gitea.io/gitea/services/mirror code.gitea.io/gitea/services/pull code.gitea.io/gitea/services/release code.gitea.io/gitea/services/repository code.gitea.io/gitea/services/wiki
# github.com/mattn/go-sqlite3
sqlite3-binding.c: In Funktion »sqlite3SelectNew«:
sqlite3-binding.c:125801:10: Warnung: Funktion liefert möglicherweise Adresse einer lokalen Variablen zurück [-Wreturn-local-addr]
125801 |   return pNew;
       |          ^~~~
sqlite3-binding.c:125761:10: Anmerkung: hier deklariert
125761 |   Select standin;
       |          ^~~~~~~
?   	code.gitea.io/gitea	[no test files]
ok  	code.gitea.io/gitea/cmd	0.277s
?   	code.gitea.io/gitea/contrib/environment-to-ini	[no test files]
?   	code.gitea.io/gitea/contrib/fixtures	[no test files]
?   	code.gitea.io/gitea/contrib/pr	[no test files]
ok  	code.gitea.io/gitea/models	66.305s
?   	code.gitea.io/gitea/models/migrations	[no test files]
?   	code.gitea.io/gitea/modules/analyze	[no test files]
ok  	code.gitea.io/gitea/modules/auth	0.187s
?   	code.gitea.io/gitea/modules/auth/ldap	[no test files]
?   	code.gitea.io/gitea/modules/auth/oauth2	[no test files]
ok  	code.gitea.io/gitea/modules/auth/openid	1.016s
?   	code.gitea.io/gitea/modules/auth/pam	[no test files]
?   	code.gitea.io/gitea/modules/auth/sso	[no test files]
ok  	code.gitea.io/gitea/modules/avatar	0.149s
ok  	code.gitea.io/gitea/modules/base	0.136s
?   	code.gitea.io/gitea/modules/cache	[no test files]
ok  	code.gitea.io/gitea/modules/charset	0.189s
ok  	code.gitea.io/gitea/modules/context	0.310s
ok  	code.gitea.io/gitea/modules/convert	0.829s
?   	code.gitea.io/gitea/modules/cron	[no test files]
ok  	code.gitea.io/gitea/modules/emoji	0.034s
ok  	code.gitea.io/gitea/modules/eventsource	0.239s
ok  	code.gitea.io/gitea/modules/generate	0.008s
ok  	code.gitea.io/gitea/modules/git	0.936s
?   	code.gitea.io/gitea/modules/git/pipeline	[no test files]
ok  	code.gitea.io/gitea/modules/gitgraph	0.122s
?   	code.gitea.io/gitea/modules/graceful	[no test files]
?   	code.gitea.io/gitea/modules/highlight	[no test files]
?   	code.gitea.io/gitea/modules/httplib	[no test files]
ok  	code.gitea.io/gitea/modules/indexer/code	1.157s
ok  	code.gitea.io/gitea/modules/indexer/issues	5.819s
ok  	code.gitea.io/gitea/modules/indexer/stats	5.485s
?   	code.gitea.io/gitea/modules/lfs	[no test files]
ok  	code.gitea.io/gitea/modules/log	0.251s
ok  	code.gitea.io/gitea/modules/markup	1.610s
?   	code.gitea.io/gitea/modules/markup/common	[no test files]
ok  	code.gitea.io/gitea/modules/markup/csv	0.169s
?   	code.gitea.io/gitea/modules/markup/external	[no test files]
ok  	code.gitea.io/gitea/modules/markup/markdown	0.261s
ok  	code.gitea.io/gitea/modules/markup/mdstripper	0.169s
ok  	code.gitea.io/gitea/modules/markup/orgmode	0.222s
?   	code.gitea.io/gitea/modules/metrics	[no test files]
ok  	code.gitea.io/gitea/modules/migrations	7.930s
?   	code.gitea.io/gitea/modules/migrations/base	[no test files]
?   	code.gitea.io/gitea/modules/notification	[no test files]
ok  	code.gitea.io/gitea/modules/notification/action	0.567s
?   	code.gitea.io/gitea/modules/notification/base	[no test files]
?   	code.gitea.io/gitea/modules/notification/indexer	[no test files]
?   	code.gitea.io/gitea/modules/notification/mail	[no test files]
?   	code.gitea.io/gitea/modules/notification/ui	[no test files]
?   	code.gitea.io/gitea/modules/notification/webhook	[no test files]
?   	code.gitea.io/gitea/modules/options	[no test files]
ok  	code.gitea.io/gitea/modules/password	0.301s
?   	code.gitea.io/gitea/modules/pprof	[no test files]
?   	code.gitea.io/gitea/modules/private	[no test files]
ok  	code.gitea.io/gitea/modules/process	0.247s
?   	code.gitea.io/gitea/modules/public	[no test files]
ok  	code.gitea.io/gitea/modules/queue	1.564s
?   	code.gitea.io/gitea/modules/recaptcha	[no test files]
ok  	code.gitea.io/gitea/modules/references	0.186s
ok  	code.gitea.io/gitea/modules/repofiles	6.069s
ok  	code.gitea.io/gitea/modules/repository	1.512s
ok  	code.gitea.io/gitea/modules/secret	0.009s
?   	code.gitea.io/gitea/modules/session	[no test files]
ok  	code.gitea.io/gitea/modules/setting	0.130s
?   	code.gitea.io/gitea/modules/ssh	[no test files]
?   	code.gitea.io/gitea/modules/structs	[no test files]
?   	code.gitea.io/gitea/modules/svg	[no test files]
ok  	code.gitea.io/gitea/modules/sync	0.009s
?   	code.gitea.io/gitea/modules/task	[no test files]
ok  	code.gitea.io/gitea/modules/templates	0.292s
?   	code.gitea.io/gitea/modules/test	[no test files]
ok  	code.gitea.io/gitea/modules/timeutil	0.155s
ok  	code.gitea.io/gitea/modules/upload	0.008s
ok  	code.gitea.io/gitea/modules/user	0.050s
ok  	code.gitea.io/gitea/modules/util	0.176s
ok  	code.gitea.io/gitea/modules/validation	0.127s
ok  	code.gitea.io/gitea/modules/webhook	0.744s
?   	code.gitea.io/gitea/routers	[no test files]
ok  	code.gitea.io/gitea/routers/admin	1.934s
?   	code.gitea.io/gitea/routers/api/v1	[no test files]
?   	code.gitea.io/gitea/routers/api/v1/admin	[no test files]
ok  	code.gitea.io/gitea/routers/api/v1/misc	0.255s
?   	code.gitea.io/gitea/routers/api/v1/notify	[no test files]
?   	code.gitea.io/gitea/routers/api/v1/org	[no test files]
ok  	code.gitea.io/gitea/routers/api/v1/repo	1.469s
?   	code.gitea.io/gitea/routers/api/v1/settings	[no test files]
?   	code.gitea.io/gitea/routers/api/v1/swagger	[no test files]
?   	code.gitea.io/gitea/routers/api/v1/user	[no test files]
?   	code.gitea.io/gitea/routers/api/v1/utils	[no test files]
?   	code.gitea.io/gitea/routers/dev	[no test files]
?   	code.gitea.io/gitea/routers/events	[no test files]
?   	code.gitea.io/gitea/routers/org	[no test files]
?   	code.gitea.io/gitea/routers/private	[no test files]
ok  	code.gitea.io/gitea/routers/repo	12.761s
?   	code.gitea.io/gitea/routers/routes	[no test files]
ok  	code.gitea.io/gitea/routers/user	0.936s
ok  	code.gitea.io/gitea/routers/user/setting	4.045s
ok  	code.gitea.io/gitea/routers/utils	0.006s
?   	code.gitea.io/gitea/services/comments	[no test files]
?   	code.gitea.io/gitea/services/externalaccount	[no test files]
ok  	code.gitea.io/gitea/services/gitdiff	1.326s
ok  	code.gitea.io/gitea/services/issue	1.279s
ok  	code.gitea.io/gitea/services/mailer	0.930s
ok  	code.gitea.io/gitea/services/mirror	1.376s
ok  	code.gitea.io/gitea/services/pull	0.943s
ok  	code.gitea.io/gitea/services/release	12.994s
ok  	code.gitea.io/gitea/services/repository	0.483s
ok  	code.gitea.io/gitea/services/wiki	1.622s
```

### CODE REVIEW

**Comment by no-go** [comment]
Please help! I did not know how to fix the "lint-backend swagger-check": it dislike changes on `templates/swagger/v1_json.tmpl`

I did not find a solution to "extend" the api with a new version. Should I only add a new `v2_json.tmpl` ?

**Comment by lunny** [comment]
Did you run `make generate-swagger`?

File: models/issue_stopwatch.go [hunk 1]
```diff
@@ -201,9 +201,16 @@ func (sw *Stopwatch) APIFormat() (api.StopWatch, error) {
 	if err != nil {
 		return api.StopWatch{}, err
 	}
+	repo, err := getRepositoryByID(x, issue.RepoID)
+	if err != nil {
+		return api.StopWatch{}, err
+	}
```
Comment (zeripath) [inline models/issue_stopwatch.go]:
```suggestion
	if err := issue.LoadRepo(); err != nil {
		return api.StopWatch{}, err
	}
```

File: models/issue_stopwatch.go [hunk 2]
```diff
@@ -201,9 +201,16 @@ func (sw *Stopwatch) APIFormat() (api.StopWatch, error) {
...
 	return api.StopWatch{
-		Created:    sw.CreatedUnix.AsTime(),
-		IssueIndex: issue.Index,
+		Created:       sw.CreatedUnix.AsTime(),
+		IssueIndex:    issue.Index,
+		IssueTitle:    issue.Title,
+		RepoOwnerName: repo.OwnerName,
+		RepoName:      repo.Name,
```
Comment (zeripath) [inline models/issue_stopwatch.go]:
```suggestion
		RepoOwnerName: issue.Repo.OwnerName,
		RepoName:      issue.Repo.Name,
```

**Comment by zeripath** [comment]
It's probably also sensible to cache the repo in the api.StopWatches level

**Comment by 6543** [comment]
@zeripath this could be done when we move APIFormat() to the **convert** package and add a function to convert model.StopWatchList's ...

**Review by 6543 — APPROVED** [review]

**Review by zeripath — APPROVED** [review]

**Comment by 6543** [comment]
as montioned the refactor:  #12729

**Pull request merged**